.github/workflows/macos.yml

```diff
@@ -0,0 +1,18 @@
+name: macOS
```

Comment:
In my experience, it takes a while to provision a macOS machine, which gets in the way of actually merging PRs. They're also relatively expensive (10x the cost of Linux minutes). Can we instead run this job 1x/week or similar?

Reply:
Switched to run twice a week.

I was under the impression that workflow minutes were free for opensource? In either case I understand desire to avoid macOS in the PR workflow as it can be frustratingly slow / unreliable.